### ISSUE TITLE
[saas-3] CRM: Constraint errors when merging partners

### ISSUE THREAD

**Issue by JonathanNEMRY** [odoo/odoo#9705]
Impacted versions:

 - saas-3
 - previously on 8.0 but fixed into https://github.com/odoo/odoo/commit/8d23a3a86c3fc09e0b9f6275641639106657b2e7 by @mart-e 

Steps to reproduce:

 1. Set same email for two partner
 2. Follow these partners
 3. Go into sale/Tools/Deduplicate Contacts 
 4. Merge 

Current behavior: 

```
Constraint Error

Error, a partner cannot follow twice the same object
```

Expected behavior:

 - No constraint Error
@mart-e could you please backport your fix?

Thanks!

Jonathan

**Comment by JonathanNEMRY** [comment]
OPW 657338

**Comment by dmo-odoo** [comment]
Fix from @mart-e backported in saas-3 at 1660daf2fb19c5a99bb4e8df05c5bf7721e45cf8.

**Comment by JonathanNEMRY** [comment]
Thanks!
